lib/internal/modules/esm/hooks.js

```diff
@@ -403,11 +403,7 @@ class Hooks {
 
     let responseURLObj;
     if (typeof responseURL === 'string') {
-      try {
-        responseURLObj = new URL(responseURL);
-      } catch {
-        // responseURLObj not defined will throw in next branch.
-      }
+      responseURLObj = URL.parse(responseURL);
```

Comment:
Do we have access to the Ada binding directly? If so, we might prefer that as it can't be affected by user mutations

Reply:
Another option is to extract the static method as an internal function and use it.

Reply:
if I export `URLParse: URL.parse` in `internal/url.js` would it make it immutable?

Reply:
From what I can see in the implementation code, it seems we could simply pass `kParseURLSymbol` as a third argument, although `URLParse` is probably better for readability